src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedStr = '';
+    let sep = separator;
+
+    if (separator === null) {
+      sep = 'null';
+    }
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 1) {
+      return `${this[0]}`;
+    }
+
+    function value(val) {
+      if ((val === undefined) || (val === null)) {
+        return '';
```

Comment:
i would better name this function other way, you can aks question what this function does? - checkValue, or checkTypeOfValue

Reply:
thank you, fixed!